### PR TITLE
Added usage_guide.ipynb

### LINKED ISSUE #15: Create `usage_guide.ipynb` & `occurrences_ocean_sunfish.ipynb` notebooks
Should the tutorial be something added to the readme or a jupyter notebook?

### PR DESCRIPTION
**Pull request by ayushanand18**
## Overview

This PR aims to resolve #15 partly by
+ adding the `usage_guide.ipynb` notebook containing examples for prospective pyobis users.

## Notes for Reviewers
This PR is two commits ahead of #26 (one change ahead).

## Next Steps
I will now be working on creating Jupyter Notebooks for analyzing and visualizing data grabbed through occurrences.search()

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+- Discussing the current state of the code
+- Submitting a fix
+- Proposing new features
+- Becoming a maintainer
+
+## We Develop with Github
+We use github to host code, to track issues and feature requests, as well as accept pull requests.
+
+## We Use [Github Flow](https://guides.github.com/introduction/flow/index.html), So All Code Changes Happen Through Pull Requests
+Pull requests are the best way to propose changes to the codebase (we use [Github Flow](https://guides.github.com/introduction/flow/index.html)). We actively welcome your pull requests:
```
Comment (ocefpaf) [inline CONTRIBUTING.md:14]:
```suggestion
Pull requests are the best way to propose changes to the codebase (we use [GitHub Flow](https://guides.github.com/introduction/flow/index.html)). We actively welcome your pull requests:
```

**Comment by ocefpaf** [comment]
~~@ayushanand18 if you can split your PRs it will make reviewing them a bit easier.~~

---

Edit: Oh, I just saw #26 and it seems that you just pull those commits here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Contributing to pyobis
+We love your input! We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a bug
+- Discussing the current state of the code
+- Submitting a fix
+- Proposing new features
+- Becoming a maintainer
+
+## We Develop with Github
```
[the remark this answers is not in the document]
Comment (ayushanand18) [reply CONTRIBUTING.md:10]: Thanks @ocefpaf for the review! I will update these corrections.

**Comment by ayushanand18** [comment]
> Edit: Oh, I just saw #26 and it seems that you just pull those commits here.

This PR is actually some commits ahead of #26, I thought of basing this branch on the previous so that there won't be merge conflicts in the future (I had been witnessing quite a few while creating this PR).

**Comment by ocefpaf** [comment]
> I thought of basing this branch on the previous so that there won't be merge conflicts in the future (I had been witnessing quite a few while creating this PR).

The separation of context helps to avoid conflicts. If this PR only edits the notebook and the other one the markdown contributing file, we should be OK.

**Comment by ayushanand18** [comment]
> 1. We don't need the same results printed twice. A table is probably easier to visualize. Also, don't print pandas tables, let the HTML render do its magic.

I will update the notebook accordingly. Actually, for some results a pandas DataFrame doesn't speak much due to the huge volume of columns and few rows. I am thinking of keeping the raw response for some (cleaned up as a dictionary) and pandas DataFrame for some.

> 2. Try to filter out the results to make them easier to understand. Maybe create a map/graph with the data,
> 3. Try to tell a story with the example, it can be either about the API or the data itself, but something that users will find useful and that they can modify for their use cases.

I was thinking to keep this notebook only for explaining about the functions inside modules, and their input parameters. Analyzing data through each function might make this notebook bit heavy, so I thought of diving deep into each module in a separate notebook. How should I approach this?

**Comment by ayushanand18** [comment]
> The separation of context helps to avoid conflicts. If this PR only edits the notebook and the other one the markdown contributing file, we should be OK.

Noted. I'll make sure about this for future. 
Should I pull out the changes of the markdown contributing guide or update the changes here and close #26?

**Comment by ocefpaf** [comment]
> Should I pull out the changes of the markdown contributing guide or update the changes here and close #26?

Wait for @7yl4r input. I'm in favor of the easiest path here.My comment was more for future PRs.

**Comment by 7yl4r** [comment]
Uh oh. It looks like `CONTRIBUTING.md` got rolled in with this PR but #26 is specifically for that. We should either:

1. close that PR and just use this one
2. rm CONTRIBUTING.md from this PR and implement Felipe's changes over on #26.

🩹

**Comment by 7yl4r** [comment]
> This PR is actually some commits ahead of #26, I thought of basing this branch on the previous so that there won't be merge conflicts in the future (I had been witnessing quite a few while creating this PR).

I think the best approach here would have been to base a new fork off of the latest in the master. One critical piece that may be missing is using `git rebase` to keep forks up-to-date. As a general rule you should never merge-pull from master on a PR fork. I am not 100% sure, but I think [this commit](https://github.com/ayushanand18/pyobis/commit/8373a873ddb6c6a9c01e332529715745cfc398ef) should have been a rebase instead of a merge.

**Comment by 7yl4r** [comment]
> I feared that it might result into lost commits and then merge conflicts. I'll surely use `git pull --rebase` for future pulls.

It *is* something to be careful with, since rebasing does edit history. It makes me nervous too, but unfortunately it seems to be essential to the typical git PR workflow.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  - Give sample code if you can, so that *anyone* with the base python setup can reproduce. 
+- What you expected would happen
+- What actually happens
+- Notes (possibly including why you think this might be happening, or stuff you tried that didn't work)
+
+## License
+By contributing, you agree that your contributions will be licensed under [this License](https://github.com/iobis/pyobis/blob/master/LICENSE).
+
+## References
+This document was adapted from the open-source contribution guidelines for [Facebook's Draft](https://github.com/facebook/draft-js/blob/a9316a723f9e918afde44dea68b5f9f39b7d9b00/CONTRIBUTING.md)
```
[the remark this answers is not in the document]
Comment (ayushanand18) [reply CONTRIBUTING.md]: Thanks @ocefpaf for the resource! I'll get some insights from there and update the document.

**Review by 7yl4r — APPROVED** [review]

**Comment by ayushanand18** [comment]
> It _is_ something to be careful with, since rebasing does edit history. It makes me nervous too, but unfortunately it seems to be essential to the typical git PR workflow.

I'm learning new things every day, and this journey hasn't been so good before. And I have so much more to learn. Thank you so much!

**Pull request merged**